app/Http/Controllers/CourseController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Course;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Auth;
+
+class CourseController extends Controller
+{
+    public function index(): View
+    {
+        return view('courses.index', [
+            'courses' => Course::all(),
+        ]);
+    }
+
+    public function show(Course $course): View
+    {
+        $user = Auth::user();
+        $courseUser = $user->courses->find($course->id);
+        $finishedCourse = $courseUser?->getRelationValue('pivot')->finished_at;
+        $receivedCertificate = $courseUser?->getRelationValue('pivot')->received_certificate_at;
```

Comment:
I'm not exactly sure what the `getRelationValue` method does here. Could this be written as the following?

```suggestion
        $finishedCourse = $courseUser?->pivot->finished_at;
        $receivedCertificate = $courseUser?->pivot->received_certificate_at;
```

Reply:
I used getRelationValue to get around the composer analysis error saying that pivot is not defined for relationships. This method was also suggested by Ned.